app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -87,12 +91,25 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
 
             windowFeature.set(
                 feature = WindowFeature(
-                    store = context.components.core.store,
-                    tabsUseCases = context.components.useCases.tabsUseCases
+                    store = components.core.store,
+                    tabsUseCases = components.useCases.tabsUseCases
                 ),
                 owner = this,
                 view = view
             )
+            searchFeature.set(
+                feature = SearchFeature(components.core.store) {
+                    if (it.isPrivate) {
+                        components.browsingModeManager.mode = BrowsingMode.Private
```

Comment:
I don't think we need to change the browsing modes here, opening in a normal/private tab _should_ already adjust